ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -541,64 +553,112 @@ def run_ocs_upgrade(operation=None, *operation_args, **operation_kwargs):
             upgrade=True
         )
         log.info(f"Disconnected upgrade - new image: {upgrade_ocs.ocs_registry_image}")
+
+    channel = upgrade_ocs.set_upgrade_channel()
     with CephHealthMonitor(ceph_cluster):
-        channel = upgrade_ocs.set_upgrade_channel()
         upgrade_ocs.set_upgrade_images()
-        if upgrade_version != "4.9":
-            # In the case of upgrade to ODF 4.9, the ODF operator should upgrade
-            # OCS automatically.
-            upgrade_ocs.update_subscription(channel)
-        if original_ocs_version == "4.8" and upgrade_version == "4.9":
-            deployment = Deployment()
-            deployment.subscribe_ocs()
-        if (config.ENV_DATA["platform"] == constants.IBMCLOUD_PLATFORM) and not (
-            upgrade_in_current_source
-        ):
-            create_ocs_secret(config.ENV_DATA["cluster_namespace"])
-            for attempt in range(2):
-                # We need to do it twice, because some of the SA are updated
-                # after the first load of OCS pod after upgrade. So we need to
-                # link updated SA again.
-                log.info(
-                    f"Sleep 1 minute before attempt: {attempt+1}/2 "
-                    "of linking secret/SAs"
-                )
-                time.sleep(60)
-                link_all_sa_and_secret_and_delete_pods(
-                    constants.OCS_SECRET, config.ENV_DATA["cluster_namespace"]
+        ui_upgrade_supported = False
+        if config.UPGRADE.get("ui_upgrade"):
+            ocp_version = get_ocp_version()
+            if (
+                Version.coerce(ocp_version) == Version.coerce("4.9")
```

Comment:
```suggestion
                version.get_semantic_ocp_version_from_config() == version.VERSION_4_9
```
we can use version module here : https://github.com/red-hat-storage/ocs-ci/blob/f1674710eb69117d9e1d5a262740c292f4ac6d67/ocs_ci/utility/version.py

Reply:
Aman is getting actual version of the cluster from get_ocp_version which actually taking actual version not one which is passed via --ocp-version.  So to get it with this approach I suggest here to change to:

```
version.get_semantic_version(ocp_version, True) ==  version.VERSION_4_9
```
This should keep the logic and will work with compare to version.VERSION_4_9 - as this one is generated with only_major_minor = True.